### PR TITLE
feat!: upgrade to CDK v2

### LINKED ISSUE #37: CDK v2 support
We should migrate to `aws-cdk-lib` from `monocdk` when appropriate. This would likely result in a v1 release of this library.

### PR DESCRIPTION
**Pull request by echeung-amzn**
Closes #37

~~This is actually blocked by #48, so we should address that separately first.~~

Of note:

- CDK v2 doesn't bundle experimental APIs in, so they're separate `@aws-cdk` modules. This makes the peer dependencies quite awkward.
- `dimensions` is no longer accessible in the `Metric` props, forcing us to use `dimensionsMap`, which is typed to only allow `string` values. ~~I'm not entirely sure if forcibly putting non-string values works properly.~~
- The `@aws-cdk/aws-secretsmanager:parseOwnedSecretName` feature flag is removed in CDK v2.
- It's otherwise mostly just import changes

Relevant docs:

- https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: lib/monitoring/aws-dynamo/DynamoTableMetricFactory.ts [hunk 1]
```diff
@@ -98,7 +98,7 @@ export class DynamoTableMetricFactory {
       'MetricName="SuccessfulRequestLatency"',
       {
         TableName: this.table.tableName,
-        Operation: undefined,
+        Operation: undefined as unknown as string,
```
Comment (ayush987goyal) [inline lib/monitoring/aws-dynamo/DynamoTableMetricFactory.ts:101]: We would need to see if deployment passes with this. Could you try to deploy this change from a test package to see if everything works fine?

**Review by ayush987goyal — APPROVED** [review]
Overall the change looks good to me. Just one comment about testing once by deploying the change (similar to the internal test package).

Thanks a lot for working on this big change!!

File: lib/monitoring/aws-dynamo/DynamoTableMetricFactory.ts (hunk 1 above)
Comment (echeung-amzn) [reply lib/monitoring/aws-dynamo/DynamoTableMetricFactory.ts:101]: I deployed it to a test account and it seems to work fine. This is the metric that ends up in the dashboard:

```
    "metrics": [
        [ { "label": " ", "expression": "SEARCH('{AWS/DynamoDB,TableName,Operation} TableName=\"SampleTable\" MetricName=\"SuccessfulRequestLatency\"', 'Average', 300)", "region": "us-east-1" } ]
    ],
```

Which matches up with what's generated with the older monocdk-based codebase.

**Review by ayush987goyal — APPROVED** [review]

**Review by voho — APPROVED** [review]
LGTM! However, there might still be some changes that are not covered, I think it might be the case of WAF.

**Comment by echeung-amzn** [comment]
> there might still be some changes that are not covered, I think it might be the case of WAF.

You mean in terms of pre-existing code coverage?

**Comment by voho** [comment]
> 

I mean that these would need a migration to CDK v2, since they include import changes, right?

**Comment by echeung-amzn** [comment]
This has been rebased on top of the current `HEAD`, so I'll go ahead and allow this to be merged.

**Pull request merged**